packages/@coorpacademy-components/src/molecule/skill-picker-modal/index.js

```diff
@@ -0,0 +1,157 @@
+
+  const handleCancel = useCallback(() => {
+    setSelectedSkillList(selectedSkills);
+    onCancel();
+  }, [setSelectedSkillList, selectedSkills, onCancel]);
+
+  const handleClose = useCallback(() => {
+    setSelectedSkillList(selectedSkills);
+    onClose();
+  }, [setSelectedSkillList, selectedSkills, onClose]);
+
+  const skillList = useMemo(() => {
+    return skills.map(skill => {
+      return {
+        skillTitle: skillsLocales[skill],
+        skillRef: skill,
+        focus: selectedSkillList.includes(skill)
+      };
+    });
+  }, [skills, selectedSkillList, skillsLocales]);
+
+  const isError = useMemo(
+    () => selectedSkillList.length > maxSelectedSkills,
+    [selectedSkillList, maxSelectedSkills]
+  );
+
+  const footer = useMemo(() => {
+    const footerDescription = isError
+      ? translate('skill_focus_footer_error_description', {
+          skillNumber: selectedSkillList.length - maxSelectedSkills
+        })
+      : translate('skill_focus_footer_description', {
+          skillNumber: minSelectedSkills - selectedSkillList.length
+        });
+    return {
+      text:
+        isLoading ||
+        (selectedSkillList.length <= maxSelectedSkills &&
+          selectedSkillList.length >= minSelectedSkills)
+          ? ''
+          : footerDescription.replace('.'),
+      isError,
+      cancelButton: {
+        onCancel: handleCancel,
+        label: translate('cancel'),
+        disabled: isLoading || isError
+      },
+      confirmButton: {
+        onConfirm: () => onConfirm(selectedSkillList),
+        label: translate('confirm'),
+        iconName: 'circle-check',
+        disabled: isLoading || isError
+      }
+    };
+  }, [
+    isError,
+    handleCancel,
+    onConfirm,
+    translate,
+    selectedSkillList,
+    isLoading,
+    maxSelectedSkills
+  ]);
+
+  if ((!isLoading && !skills) || !isOpen) return null;
+
+  return (
+    <BaseModal
+      title={translate('skill_focus')}
+      description={translate('skill_focus_description')}
+      isOpen={isOpen}
+      onClose={handleClose}
+      footer={footer}
+      headerIcon={{
+        name: 'bullseye-arrow',
+        backgroundColor: '#DDD1FF'
+      }}
+    >
+      <div className={style.skillPickerContainer}>
+        {isLoading ? (
+          <div className={style.loaderContainer}>
+            <Loader className={style.loader} theme="coorpmanager" />
+          </div>
+        ) : (
+          <>
+            <div style={{marginBottom: '16px'}}>
+              {`${selectedSkillList.length} ${translate('selected')}`}
+            </div>
+            <div style={{display: 'flex', gap: '16px', flexWrap: 'wrap'}}>
+              {skillList.map((skill, index) => {
+                const {skillTitle, focus} = skill;
+                function handleChipClick() {
+                  let filteredSelectedSkillList = [...selectedSkillList];
+                  if (selectedSkillList.includes(skill.skillRef)) {
```

Comment:
same as ?: 
```suggestion
                  if (skill.focus)) {
```

Reply:
Ah right, Thank you Adam, I have updateed code to resolve all the comment above also.